tensorflow/python/estimator/canned/tensor_forest.py

```diff
@@ -0,0 +1,257 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Estimator classes for TensorForest."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import collections
+import math
+
+from tensorflow.python.estimator import estimator
+from tensorflow.python.estimator.canned import head as head_lib
+from tensorflow.python.ops.losses import losses
+from tensorflow.python.summary import summary
+from tensorflow.python.feature_column import feature_column as feature_column_lib
+from tensorflow.python.ops import gen_tensor_forest_ops
+from tensorflow.python.ops import tensor_forest_ops, math_ops, array_ops
+
+# from tensorflow.python.util.tf_export import estimator_export
+
+_ForestHParams = collections.namedtuple('TreeHParams', [
+    'logits_dimension',
+    'n_trees', 'max_nodes', 'num_splits_to_consider',
+    'split_node_after_samples', 'is_regression',
+])
+
+VARIANCE_PREDICTION_KEY = 'prediction_variance'
+
+
+class RandomForestGraphs(object):
```

Comment:
should we be using RandomForest or tensorForest. It is not a classical random forest...